content/docs/rfcs/36/README.md

```diff
@@ -683,6 +785,202 @@ For example:
 ```
 
 
+## Waku Filter
+
+### `extern char* waku_filter_subscribe(char* filterJSON, char* peerID, int ms)`
+
+Creates a subscription in a lightnode for messages that matches a content filter and optionally a [PubSub `topic`](https://github.com/libp2p/specs/blob/master/pubsub/README.md#the-topic-descriptor).
+
+**Parameters**
+
+1. `char* filterJSON`: JSON string containing the [`FilterSubscription`](#filtersubscription-type) to unsubscribe from
+2. `char* peerID`: Peer ID to subscribe to.
+   The peer must be already known.
+   It must have been added before with [`waku_add_peer`](#extern-char-waku_add_peerchar-address-char-protocolid)
+   or previously dialed with [`waku_connect_peer`](#extern-char-waku_connect_peerchar-address-int-timeoutms).
+   Use `NULL` to automatically select a node.
+3. `int timeoutMs`: Timeout value in milliseconds to execute the call.
+   If the function execution takes longer than this value,
+   the execution will be canceled and an error returned.
+   Use `0` for no timeout.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse-type).
+If the execution is successful, the `result` field is set to `true`.
+
+For example:
+
+```json
+{
+  "result": true
+}
+```
+
+**Events**
+
+When a message is received, a ``"message"` event` is emitted containing the message, pubsub topic, and node ID in which
+the message was received.
+
+The `event` type is [`JsonMessageEvent`](#jsonmessageevent-type).
+
+For Example:
+
+```json
+{
+  "type": "message",
+  "event": {
+    "subscriptionID": 1,
+    "pubsubTopic": "/waku/2/default-waku/proto",
+    "messageID": "0x6496491e40dbe0b6c3a2198c2426b16301688a2daebc4f57ad7706115eac3ad1",
+    "wakuMessage": {
+      "payload": "TODO",
+      "contentTopic": "/my-app/1/notification/proto",
+      "version": 1,
+      "timestamp": 1647826358000000000
+    }
+  }
+}
+```
+
+### `extern char* waku_filter_unsubscribe(char* filterJSON, int timeoutMs)`
+
+Removes subscriptions in a light node matching a content filter and, optionally, a [PubSub `topic`](https://github.com/libp2p/specs/blob/master/pubsub/README.md#the-topic-descriptor).
```

Comment:
Is the idea that it unsubscribes from all peers where a matching filter exists? Asking because `subscribe` can be peer-specific, but `unsubscribe` is not.

Reply:
Yes, it would try to remove from all peers where such subscription exists. I tried to make the API similar to [16/WAKU2-RPC
](https://rfc.vac.dev/spec/16/#delete_waku_v2_filter_v1_subscription)